### DISCUSSION TITLE
max_length validation is not wroking with Modelserializer

### DISCUSSION THREAD

**Discussion by Anshulgoplus** [encode/django-rest-framework#9489] [Question & Answer] [answered]
I created a model have a field as charField and has max_length then i created a model serailzer for that model by the serializer is not validating the field with max_length

class MyModel(models.Model):
              name = models.CharField(max_length=255, verbose_name=_("Name"))

class MyModelSerializer(ModelSerializer):
             classs Meta:
                       model=MyModel
                       fields = "__all__"
                       
when I send a post request with name as 256 chars it is throwing 500 it should throw 400

**Comment by smchougrani** [comment]
Hi Anshulgoplus,

Thank you for raising this issue. The problem you described involves the validation of the maximum length of a `CharField` in a `ModelSerializer`. When you send a POST request with a `name` field containing 256 characters, you receive a 500 error instead of the expected 400 error.

**Issue Analysis**
The issue seems to stem from the `ModelSerializer` not automatically enforcing the maximum length constraint defined in the model. To resolve this, you can explicitly define the `name` field in the serializer with the `max_length` constraint.

**Proposed Solution**
Here’s how you can modify your serializer to include this validation:

```python
from rest_framework import serializers

class MyModelSerializer(serializers.ModelSerializer):
    name = serializers.CharField(max_length=255)

    class Meta:
        model = MyModel
        fields = '__all__'
```

**Explanation**
By explicitly defining the `name` field in the serializer with `max_length=255`, you ensure that the maximum length validation is applied when the data is deserialized. This will raise a `ValidationError` if the length of the string exceeds 255 characters, resulting in a 400 error instead of a 500 error.

**Conclusion**
This solution should resolve the issue without requiring changes to the core framework. If you have any further questions or encounter other issues, please feel free to share them.

Thank you again for your contribution!

Best regards,
Chougrani Sidi Mohammed

**Comment by humayunah** [comment, marked as answer]
@Anshulgoplus I can't seem to replicate this issue with the latest version of djangorestframework and without explicitly defining the name field in the serializer as suggested above, I got the following validation error with status code 400.
```
{
    "name": [
        "Ensure this field has no more than 255 characters."
    ]
}
```

**Comment by smchougrani** [comment]
Hi @Anshulgoplus,

Thank you for your prompt feedback. It's great to hear that you couldn't replicate the issue with the latest version of Django REST Framework and that the validation error is being correctly handled with a status code 400.
